syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

message OpenSearch {

  message QueryLogs {
    google.protobuf.StringValue index = 1;
    google.protobuf.StringValue query_dsl = 2;
    google.protobuf.UInt64Value limit = 3;
    google.protobuf.UInt64Value offset = 4;
    google.protobuf.StringValue sort_desc = 5;
    google.protobuf.StringValue timestamp_field = 6;
  }

  message DeleteIndex {
    google.protobuf.StringValue index = 1;
  }

  message GetNodeStats {
  }

  message GetIndexStats {
  }

  message GetClusterHealth {
  }

  message GetClusterSettings {
  }

  message GetClusterStats {
  }

  message GetIndexAndShardRecoveries {
  }

  message GetIndices {
  }

  message GetPendingTasks {
  }

  message GetShards {
  }

  message GetTasks {
  }

  enum TaskType {
    UNKNOWN = 0;
    QUERY_LOGS = 1;
    DELETE_INDEX = 2;
    GET_NODE_STATS = 3;
    GET_INDEX_STATS = 4;
    GET_CLUSTER_HEALTH = 5;
    GET_CLUSTER_SETTINGS = 6;
    GET_CLUSTER_STATS = 7;
    GET_INDEX_AND_SHARD_RECOVERIES = 8;
    GET_INDICES = 9;
    GET_PENDING_TASKS = 10;
    GET_SHARDS = 11;
    GET_TASKS = 12;
  }

  TaskType type = 1;
  oneof task {
    QueryLogs query_logs = 101;
    DeleteIndex delete_index = 102;
    GetNodeStats get_node_stats = 103;
    GetIndexStats get_index_stats = 104;
    GetClusterHealth get_cluster_health = 105;
    GetClusterSettings get_cluster_settings = 106;
    GetClusterStats get_cluster_stats = 107;
    GetIndexAndShardRecoveries get_index_and_shard_recoveries = 108;
    GetIndices get_indices = 109;
    GetPendingTasks get_pending_tasks = 110;
    GetShards get_shards = 111;
    GetTasks get_tasks = 112;
  }
}